src/session.rs

```diff
@@ -270,20 +270,89 @@ impl Session {
     ///
     /// If `program` is not an absolute path, the `PATH` will be searched in an OS-defined way on
     /// the host.
-    pub fn raw_command<S: AsRef<OsStr>>(&self, program: S) -> Command<'_> {
-        Command::new(
-            self,
-            delegate!(&self.0, imp, { imp.raw_command(program.as_ref()).into() }),
-        )
+    pub fn raw_command<S: AsRef<OsStr>>(&self, program: S) -> OwningCommand<&'_ Self> {
+        Self::to_raw_command(self, program)
+    }
+
+    /// Version of [`command`](Self::command) which stores an
+    /// `Arc<Session>` instead of a reference, making the resulting
+    /// [`OwningCommand`] independent from the source [`Session`] and
+    /// simplifying lifetime management and concurrent usage:
+    ///
+    /// ```rust,no_run
+    /// # use std::sync::Arc;
+    /// # use tokio::io::AsyncReadExt;
+    /// # use openssh::{Session, KnownHosts};
+    /// # #[cfg(feature = "native-mux")]
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), openssh::Error> {
+    ///
+    /// let session = Arc::new(Session::connect_mux("me@ssh.example.com", KnownHosts::Strict).await?);
+    ///
+    /// let log = session.arc_command("less").arg("+F").arg("./some-log-file").spawn().await?;
+    /// tokio::spawn(async move {
+    ///     // can move the child around
+    ///     let stdout = log.stdout().take()?;
+    ///     let buf = vec![0;100];
+    ///     while let Ok(n) = stdout.read(&mut buf).await? {
+    ///         if n == 0 {
+    ///             break;
+    ///         }
+    ///         println!("read {:?}", &buf[..n]);
+    ///     }
+    ///     Ok(())
+    /// })
+    /// # Ok(()) }
+    pub fn arc_command<'a, P: Into<Cow<'a, str>>>(
+        self: std::sync::Arc<Session>,
```

Comment:
I think taking a 

```suggestion
        self: &std::sync::Arc<Session>,
```

is better and more ergonomic, otherwise users would have to do `session.clone().arc_command(..)`

Reply:
I think the current version is better, because otherwise the arc has to be cloned internally, so if the caller just wants to create a movable command/child they can't just move the one arc they have they get additional and unnecessary refcount traffic.

Reply:
That's reasonable.